### ISSUE TITLE
rtc is not working

### ISSUE THREAD

**Issue by DR-Soaps** [spiritualized1997/openFPGA-GBA#16]
games that need rtc will not work after fist start

**Comment by PriMieon** [comment]
Will the game itself not work or the rtc?

**Comment by DR-Soaps** [comment]
Just not retaining rtc after reboot

**Comment by Kitocco** [comment]
I can add to this; I just started Pokemon Emerald and the setup works fine on launch, but as soon as you leave the game it stops taking time until the next bootup. No message of "the internal battery has run dry" shows up on startup.
At the moment it seems the Pocket's clock feature only keeps time for Memeories & screenshot time & dates.

Is there any way for these cores to hook into or read the time passage of the system, or is this something only carts with batteries can do?

**Comment by evie-gh** [comment]
Yes, cores can access the RTC, they just need to be updated to do so. I'm
not actually sure if there's going to be any additional development on
these cores, though... radio silence since the first release AFAIK

**Comment by Guzaboru** [comment]
@spiritualized1997 
I would love to see the RTC feature for GBC and GBA core. Analogue implemeted the RTC for the pocket over a year ago.
Are there any plans for additional development or is it the final release of these cores?

**Comment by k5jat** [comment]
Sadly, it's mid-October 2023, and still no RTC support. :(

**Comment by StetsonFinch** [comment]
@spiritualized1997 I’ll personally pay you 50 dollars to implement RTC. Just hit me up when you do. I’m not joking.

**Comment by k5jat** [comment]
@spiritualized1997, if you have no interest in maintaining the code, please opensource so someone can take it over? It would be much better for the community.

**Comment by carlosalban** [comment]
@spiritualized1997 happy to help contribute if you decide to open source.

**Comment by aaguirre27** [comment]
Is there any update regarding this issue ? :(

**Comment by k5jat** [comment]
There is a port of the mister gb/gbc cores to the AP. I don't know if the gba core is in the works. You can find the post to r/AnalpoguePocket on Reddit.

Edit: the new gbc port has RTC.

**Comment by awolowiecki720** [comment]
Any updates?? 

I and many others in the community would love to support RTC for this GBA core
